src/trading/ClaimProceeds.sol

```diff
@@ -0,0 +1,50 @@
+pragma solidity ^0.4.13;
+
+import 'ROOT/Controller.sol';
+import 'ROOT/reporting/Interfaces.sol';
+import 'ROOT/extensions/MarketFeeCalculator.sol';
+
+/**
+ * @title ClaimProceeds
+ * @dev This allows users to claim their money from a market by exchanging their shares 
+ */
+contract ClaimProceeds is Controlled {
+	uint256 private constant THREE_DAYS = 3 * 24 * 60 * 60;
+
+	function publicClaimProceeds(IMarket market) onlyInGoodTimes public returns(uint256) {
+		require(market.isFinalized());
+		require(block.timestamp > market.getFinalizationTime() + THREE_DAYS);
+
+	    uint8 outcome = 0;
+	    while (outcome < market.getNumberOfOutcomes()) {
+	        int256 payoutNumerator = market.getFinalWinningReportingToken().getPayoutNumerator(outcome);
+	        IShareToken shareToken = market.getShareToken(outcome);
+	        int256 balance = 100;// TODO shareToken.balanceOf(msg.sender);
+	        int256 proceedAmount = balance * market.getCompleteSetCostInAttotokens() / 10**18 * payoutNumerator / market.getPayoutDenominator();
```

Comment:
This should be using safe math functions rather than raw `*` and `/`.  You can use this by adding `using SafeMath for uint256;` to the top of the contract and then `balance.mul(...)`.